### PR TITLE
Init scrypt-shim repository

### PR DESCRIPTION
**Pull request by nivida**
The following code got copied from the web3-eth-accounts module. This repository will be used as a dependency in version 1.x and 2.x.

### CODE REVIEW

**Comment by michaelsbradleyjr** [comment]
Nice work!

`web3-eth-accounts` strings should be replaced with `scrypt-shim`, I think.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "scrypt-shim",
+  "version": "0.0.1",
+  "description": "scrypt shim",
+  "main": "src/index.js",
+  "browser": "src/browser.js",
+  "scripts": {
+    "postinstall": "node postinstall.js"
```
Comment (michaelsbradleyjr) [inline package.json]: You have the postinstall script in `src/`. So the path here needs to be adjusted to `src/postinstall.js` or maybe pull the script into the repo's root, or put it in `scripts/postinstall.js` (and adjust the path accordingly).

**Review by michaelsbradleyjr — CHANGES_REQUESTED** [review]

File: scripts/postinstall.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+// there are presently no Node.js versions with built-in scrypt that doesn't
+// have maxmem limitations
+// see: https://github.com/nodejs/node/pull/28799#issuecomment-521829488
+// anticipating improved scrypt in 10.x and 12.x release lines
+// var NODE_MIN_VER_SCRYPT_MAXMEM_FIXED = ['10.?.?', '12.?.?'];
+
+// use NODE_MIN_VER_SCRYPT_MAXMEM_FIXED instead, when such versions are available
+// var byUpgradingOrInstalling = 'You can improve the performance of web3-eth-accounts by upgrading to Node.js version ' + NODE_MIN_VER_SCRYPT_MAXMEM_FIXED.join(' or ') + ' or newer, or by installing the (deprecated) scrypt package in your project';
+
+var byUpgradingOrInstalling = 'You can improve the performance of web3-eth-accounts by upgrading to Node.js version ' + NODE_MIN_VER_WITH_BUILTIN_SCRYPT + ' or newer, or by installing the (deprecated) scrypt package in your project';
```
Comment (michaelsbradleyjr) [inline scripts/postinstall.js]: Two more swaps to `scrypt` needed (here and on L12), my apologies for forgetting to point it out in my previous review.

**Review by michaelsbradleyjr — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/web3-js/scrypt-shim.git"
+  },
+  "dependencies": {
+    "scryptsy": "^2.1.0",
+    "semver": "^6.3.0"
+  },
+  "author": "",
```
Comment (michaelsbradleyjr) [inline package.json]: Perhaps we shouldn't leave `"author"` blank? I'm not concerned about getting credit so much, just might seem strange to folks looking at it on the registry's website, etc.

Comment (nivida) [reply package.json]: I will add ``web3.js`` as an author if this is okay for you.

**Review by michaelsbradleyjr — APPROVED** [review]

**Pull request merged**